### PR TITLE
Radically simplify deployment, particularly on clusters without jq and cifs-utils

### PR DESCRIPTION
**Pull request by snickell**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
/kind feature

**What this PR does / why we need it**:

This PR makes it much easier and less error prone to deploy an SMB-backed FlexVolume, particularly for first-time deployees. For the most part, installers now mainly need to know about their SMB server, share name, and login credentials, rather than learning a lot of K8S aspects they may or may not care about.

I've tried to maintain an Azure-first stance with the PR for the default configuration, but I'm not using Azure for this project, so somebody may want to check that its still optimized for Azure configuration-wise out-of-the-box.

1. Adds a helm chart with a single configuration file for all deployment steps, and a single deploy command.
2. Optionally injects statically linked copies of jq and mount.cifs onto cluster nodes, making deploys MUCH easier on clusters that don't already have these.
3. Creates a test pod by default which reduces the error prone initial debug step.

This doesn't improve everything, it could still use improvements in two areas:
- Make creating the `smbcreds` secret easier, roll it into the "all in one" somehow
- Make it easier to get the path to the flex volume plugins for your K8S platform easier... users will probably still struggle with this, though I've documented it as well as I could.

**Special notes for your reviewer**:

**Release note**:
```

```

### CODE REVIEW

**Comment by snickell** [comment]
The easiest way to understand this PR is to read `kubernetes-volume-drivers/flexvolume/smb/helm/README.md`:

# Getting an SMB-backed PersistentVolume going using this Helm chart

You can use this chart to add an SMB Share as a Volume
to your Kubernetes cluster. It enables SMB-based PersistentVolumes
on your cluster in general, so after deploying you can create as
many SMB-backed volumes as you'd like.

By default the helm chart deploys a PersistentVolume, a PersistentVolumeClaim,
a "Test" Pod for debugging the setup, and most importantly, a DaemonSet
which can install the SMB drivers (and dependencies: jq and cifs-utils)
on each node in your K8S cluster.

## Configure the smbVolume for your SMB server/share and deploy:

1. Edit `values.yaml` and configure lines marked *MUST CONFIGURE*:
  - Set `smbVolume.server`, `smbVolume.share` to access //smbVolume.server/smbVolume.share
  - Create an `smbcreds` secret on your k8s cluster with your SMB login credentials:
   ```
   kubectl create secret generic smbcreds --from-literal username=USERNAME --from-literal password="PASSWORD" --type="microsoft.com/smb"
   ```
  - Set `smbFlexVolInstaller.flexVolumePluginPath` to the correct path
    for your kubernetes platform provider (Azure AKS, minikube, etc, 
    see https://rook.io/docs/rook/master/flexvolume.html or comments in values.yaml)

2. Deploy the chart to your k8s cluster: 
   ```
   helm install --wait ./helm/smb-flexvol
   ```

## Use the testpod to validate your setup:

We recommend experimenting with your smbVolume and smbVolumeClaim in the testPod
until everything is working how you want.

3. Check if your smbVolume mounted to testPod.mountPath (/data by default):
  ```
  kubectl exec $(kubectl get pod -l "role=testpod" -o name) -- ls /data`
  ```

4. Debug interactively on the testPod if necessary:
  ```
  kubectl exec -it $(kubectl get pod -l "role=testpod" -o name) -- bash
  ```

That's it! Happy helming!

**Comment by snickell** [comment]
I evaluated building statically linked binaries in docker by reconfiguring and rebuilding Debian packages for `jq` and `cifs-utils`. I have a version which does this, and it works, but it generates a MUCH slower and longer docker install, and is a lot messier than I would have liked. After seeing what was required (and how specific it was to the current debian package configuration) `staticx` approach actually seemed much more modular, cleaner, and less likely to bitrot with time.

**Comment by andyzhangx** [comment]
thanks for the contribution, is this PR done?
@snickell is it possible to add some basic test? currently there is only very simple test here: https://github.com/Azure/kubernetes-volume-drivers/blob/master/flexvolume/smb/test/run-test.sh

**Comment by snickell** [comment]
@andyzhangx Hi! Yes, this PR is done, I'm actually using it in production at the moment.

I'm not clear exactly what aspects of this PR would be worth the effort to write tests for, since its entirely build and infrastructure and not really any "running code", if that makes sense? Probably a separate (and fairly large) project would be to add infra/deploy tests for the whole repo using `kind` to actually create a cluster and do a test deploy to it.... basically testing infra using integration testing vs testing code with unit testing. Unfortunately that's a lot more time than I can spend on this project (its probably a bigger project than a helm chart by a lot).

**Comment by andyzhangx** [comment]
thanks, let me try to add some tests first, will review this PR later.

File: flexvolume/smb/deployment/smb-flexvol-installer/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,33 @@
...
+  pip3 install --no-cache-dir staticx && \
+  staticx `which mount.cifs` ${SOURCE_DIR}/mount.cifs && \
+  chmod +s ${SOURCE_DIR}/mount.cifs && \
+  staticx `which jq` ${SOURCE_DIR}/jq
+
+###################################################
 FROM busybox
+###################################################
+
+ARG INSTALL_DEPS=false
```
Comment (andyzhangx) [inline flexvolume/smb/deployment/smb-flexvol-installer/Dockerfile:27]: what's the default value of `INSTALL_DEPS`? in L5, it's `true`? looks like `INSTALL_DEPS` is only used in `install.sh`

Comment (snickell) [reply flexvolume/smb/deployment/smb-flexvol-installer/Dockerfile:27]: Hi @andyzhangx ! I'm not sure what L5 is? 

- INSTALL_DEPS is a new docker build argument I added, it controls whether the statically linked `jq` and `mount.cifs` are copied. 
- I set the default value to `false` (don't install statically linked dependency binaries) to replicate the existing behavior on AKS (uses the system installed jq and mount.cifs in /usr/bin, /usr/sbin), to minimize impact on existing Azure AKS users. Therefore, if you use the Docker image directly following the existing README, it'll work the same as before this PR.
- However, if you deploy using the the new Helm chart, the Helm chart overrides the Docker default ENV value for INSTALL_DEPS to be `INSTALL_DEPS=true`. So if you use Helm, you get statically linked `jq` and `mount.cifs`, but if you already deployed the low-level way, this won't change won't cause a compatibility issue.

Sorry, explaining in a confusing way, let me know if that makes sense?

Comment (snickell) [reply flexvolume/smb/deployment/smb-flexvol-installer/Dockerfile:27]: Basically:
- Deploying directly with `kubectl` (current users, users following the existing README): default is an identical deploy to before this PR (full legacy compat)
- Deploying using the Helm chart following the Helm README: default helm deploy includes static `jq` and `mount.cifs` added to the system, which makes this work smoothly on non-AKS systems (should work on AKS too).

**Pull request merged**